File: 231213/일렬로 서있는 소 2/cattle-in-a-rowing-up-2.js
const fs = require('fs')
const inputAll = () => fs.readFileSync('/dev/stdin').toString().split('\n')

let [N, arr] = inputAll();

N = Number(N);
arr = arr.split(' ').map(elm => Number(elm));
const LENGTH = arr.length;

let count = 0;

for(let i = 0; i < LENGTH; i++){
    for(let j = i + 1; j < LENGTH; j++){
        for(let k = j + 1; k < LENGTH; k++){
            if(arr[i] <= arr[j] && arr[j] < arr[k]){
                count++;
            }
        }
    }
}

console.log(count);

File: 231213/일렬로 서있는 소 2/cattle-in-a-rowing-up-2.py
N = int(input())
cowList = list(map(int, input().split()))

count = 0

for i in range(len(cowList)):
    for j in range(i + 1, len(cowList)):
        for k in range(j + 1, len(cowList)):
            if cowList[i] <= cowList[j] <= cowList[k]:
                count += 1

print(count)